### PR TITLE
Time last message

### LINKED ISSUE #14: Check time of each clients' last message
Add a variable in class Client : _time_of_last_msg with correct type
Add a checker of last time in epoll_loop
Init time of last msg in Client constructor
Update time of last msg in handle_client_event()
Define after what time we consider the client to be inactive

### PR DESCRIPTION
**Pull request by Saphirac**
Add a routine to check time of last activity of a client.
It that time exceed our defined TIMEOUT, send a PING to the client.
If the client does not respond with the correct token in response, does not update it's time of last activity and if the TIMEOUT_SINCE_PING is exceeded, disconnect the client.
Also added PING and PONG command + modified the doc.

### CODE REVIEW

File: Makefile
Comment (JonathanDUFOUR) [inline Makefile:1]: No changes except the 42 header, please revert.

**Review by JonathanDUFOUR — CHANGES_REQUESTED** [review]
Some things have been uselessly changed. Please revert that.

**Review by JonathanDUFOUR — APPROVED** [review]
Everything seems good! Great work! _(^o^)/

**Review by Saphirac — COMMENTED** [review]
Great !

**Review by gle-mini — APPROVED** [review]
Perfect :)

**Pull request merged**